src/database/database-sqlite3.cpp

```diff
@@ -191,40 +224,70 @@ MapDatabaseSQLite3::MapDatabaseSQLite3(const std::string &savedir):
 
 MapDatabaseSQLite3::~MapDatabaseSQLite3()
 {
-	FINALIZE_STATEMENT(m_stmt_read)
-	FINALIZE_STATEMENT(m_stmt_write)
-	FINALIZE_STATEMENT(m_stmt_list)
-	FINALIZE_STATEMENT(m_stmt_delete)
+	FINALIZE_STATEMENT(read)
+	FINALIZE_STATEMENT(write)
+	FINALIZE_STATEMENT(list)
+	FINALIZE_STATEMENT(delete)
 }
 
 
 void MapDatabaseSQLite3::createDatabase()
 {
-	assert(m_database); // Pre-condition
+	assert(m_database);
 
-	SQLOK(sqlite3_exec(m_database,
+	// Note: before 5.12.0 the format was blocks(pos INT, data BLOB).
+	// This function only runs for newly created databases.
+
+	const char *schema =
 		"CREATE TABLE IF NOT EXISTS `blocks` (\n"
-			"	`pos` INT PRIMARY KEY,\n"
-			"	`data` BLOB\n"
-			");\n",
-		NULL, NULL, NULL),
+			"`x` INT,"
+			"`y` INT,"
+			"`z` INT,"
+			"`data` BLOB,"
+			// Declaring a primary key automatically creates an index and the order
+			// decides which range operations can be sped up. Those are:
+			// * x BETWEEN ? AND ?
+			// * x = ? AND z BETWEEN ? AND ?
+			// * x = ? AND z = ? AND y BETWEEN ? AND ?
+			// This matches our MapSector abstraction which could come in handy.
+			"PRIMARY KEY (`x`, `z`, `y`)"
```

Comment:
add `WITHOUT ROWID`

Reply:
not a good idea according to https://www.sqlite.org/withoutrowid.html:

> WITHOUT ROWID tables work best when individual rows are not too large. [...]. That means that rows should not contain more than about 50 bytes each for a 1KiB page size or about 200 bytes each for 4KiB page size. WITHOUT ROWID tables will work (in the sense that they get the correct answer) for arbitrarily large rows [...] but traditional rowid tables tend to work faster for large row sizes. This is because rowid tables are implemented as B*-Trees [...], whereas WITHOUT ROWID tables are implemented using ordinary B-Trees [...].